lib/slim/parser.rb

```diff
@@ -336,7 +340,15 @@ def parse_tag(tag)
         # The class/id attribute is :static instead of :slim :interpolate,
         # because we don't want text interpolation in .class or #id shortcut
         syntax_error!('Illegal shortcut') unless shortcut = @attr_shortcut[$1]
-        shortcut.each {|a| attributes << [:html, :attr, a, [:static, $2]] }
+
+        if shortcut.is_a?(Proc)
+          value = shortcut.call($2)
+          attributes << [:html, :attr, value[0], [:dynamic, value[1]]]
```

Comment:
:dynamic seems wrong. As I mentioned above, we would want a loop over the returned hashmap kv pairs.